ledger/acctupdates.go

```diff
@@ -387,7 +406,13 @@ func (au *accountUpdates) lookupKv(rnd basics.Round, key string, synchronized bo
 
 		persistedData, err := au.accountsq.lookupKeyValue(key)
 		if persistedData.round == currentDbRound {
-			return persistedData.value, nil
+			if persistedData.value != nil {
```

Comment:
question: does that mean either one of following case is true here:
- `err != nil` and `persistedData.value == nil`
- `err == nil` and `persistedData.value != nil`

If not, then I would wonder if we covered all cases...

Reply:
That seems correct to me. I think these semantics are a little unusual but they're consistent with resource lookup and cache entry promotion. It seems like the goal is to make sure the cache is updated even if the err != nil.

Reply:
I think we want to writePending() whether the `value` is nil or not.  That will, effectively, mean we're caching negative responses, which seems worthwhile.

The other caches don't have a way to store a negative result, but we do (nil pointer to string).

I would change my mind if this leads to any changes in the cache itself.

Reply:
I agree, but I think we still need to guard against pushing a write to the cache when an err != nil, right?

Reply:
> we still need to guard against pushing a write to the cache when an err != nil, right?

Chiming in as a reviewer - I _think_ the desired behavior is implemented though I leave it to original commenters to verify + resolve.
* As of https://github.com/algorand/go-algorand/pull/4275/commits/25946b42b8c511286a9341ebb42657538adcfe30, cache is _not_ updated when `err != nil`.  Otherwise, it's updated.  
* I think _not_ updating when `err != nil` _is_ the desired behavior.

Reply:
yes @michaeldiamant is correct, this thread is resolving